Sources/BraintreeThreeDSecure/Public/BraintreeThreeDSecure/BTThreeDSecureRequest.h

```diff
@@ -140,6 +154,15 @@ typedef NS_ENUM(NSInteger, BTThreeDSecureShippingMethod) {
  */
 @property (nonatomic) BOOL dataOnlyRequested;
 
+/**
+ Optional. An authentication created using this property should only be used for adding a payment method to the merchant's vault and not for creating transactions.
+ Defaults to BTThreeDSecureAddCardChallengeUnspecified.
+ If set to BTThreeDSecureAddCardChallengeRequested, the authentication challenge will be requested from the issuer to confirm adding new card to the merchant's vault.
+ If set to BTThreeDSecureAddCardChallengeUnspecified and amount is 0, the authentication challenge will be presented to the user.
+ If set to BTThreeDSecureAddCardChallengeNotRequested, when the amount is 0, the authentication challenge will not be presented to the user
+ */
+@property (nonatomic) BTThreeDSecureCardAddChallenge cardAddChallengeRequested;
```

Comment:
How come on iOS we are using an enum but Android a bool? Should we keep them the same across platforms?

Reply:
Definitely agree that we should make this consistent across the sdks.

Reply:
The enum was used because we weren't certain of how to mimic the behavior of a nullable bool in objective c.

Reply:
AH you're right - we can't really do nullable bools in Objective-C. We could do some sort of custom setter on the Bool and then have a flag to detect if the value was set by the merchant or not ... but that's definitely kind of hacky.

I think the most "Objective-C" friendly route here is the enum. I think it is OK if we have to diverge from parity here to do what makes most sense for the language. _If we were in Swift, this would be as easy as `Bool?`._

But honestly, since the false case is different from the not-set case ... there technically are "3" options to this property and maybe an enum actually makes the most sense across SDKs? Thoughts?

Reply:
The Android change with nullable `Boolean` as the parameter was released this week, but we could definitely deprecate that method in favor of an enum.  I do think a `Boolean` for a param with `null`, `true`, and `false` options does make most sense from a Java language perspective, but if we want full parity on this feature we can make the change there too.

Reply:
Gotcha - yeh. Maybe we can second pass about whether or not we want to deprecate the existing Bools or just leave them as-is?